ethereum/transaction_status_task.go

```
@@ -0,0 +1,156 @@
+	ethContextInitializer func(d time.Duration) (ctx context.Context, cancelFunc context.CancelFunc),
+
+) *transactionStatusTask {
+	return &transactionStatusTask{
+		timeout:               timeout,
+		BaseTask:              transactions.BaseTask{TxService: txService},
+		ethContextInitializer: ethContextInitializer,
+	}
+}
+
+// TaskTypeName returns mintingConfirmationTaskName
+func (nftc *transactionStatusTask) TaskTypeName() string {
+	return TransactionStatusTaskName
+}
+
+// Copy returns a new instance of mintingConfirmationTask
+func (nftc *transactionStatusTask) Copy() (gocelery.CeleryTask, error) {
+	return &transactionStatusTask{
+		timeout:               nftc.timeout,
+		txHash:                nftc.txHash,
+		tenantID:              nftc.tenantID,
+		ethContextInitializer: nftc.ethContextInitializer,
+		BaseTask:              transactions.BaseTask{TxService: nftc.TxService},
+	}, nil
+}
+
+// ParseKwargs - define a method to parse CentID
+func (nftc *transactionStatusTask) ParseKwargs(kwargs map[string]interface{}) (err error) {
+	err = nftc.ParseTransactionID(kwargs)
+	if err != nil {
+		return err
+	}
+
+	tenantID, ok := kwargs[TransactionTenantIDParam].(string)
+	if !ok {
+		return errors.New("missing tenant ID")
+	}
+
+	nftc.tenantID, err = identity.CentIDFromString(tenantID)
+	if err != nil {
+		return err
+	}
+
+	// parse txHash
+	txHash, ok := kwargs[TransactionTxHashParam]
+	if !ok {
+		return errors.New("undefined kwarg " + TransactionTxHashParam)
+	}
+	nftc.txHash, ok = txHash.(string)
+	if !ok {
+		return errors.New("malformed kwarg [%s]", TransactionTxHashParam)
+	}
+
+	// override TimeoutParam if provided
+	tdRaw, ok := kwargs[queue.TimeoutParam]
+	if ok {
+		td, err := queue.GetDuration(tdRaw)
+		if err != nil {
+			return errors.New("malformed kwarg [%s] because [%s]", queue.TimeoutParam, err.Error())
+		}
+		nftc.timeout = td
+	}
+
+	return nil
+}
+
+func getTransactionStatus(ctx context.Context, txHash string) (bool, error) {
```

I think bool/err is redundant here. If we can change this to `func isTransactionSuccessful(...) bool {}`, if ok {}, then it is successful else failed
What do you think ?

---

we need to always return the error, otherwise it would be a swallowed error ;)

---

in that return an error instead of returning both is what I'm hinting at :)